src/libs/antares/study/parts/thermal/cluster_list.cpp

```diff
@@ -323,12 +326,23 @@ bool ThermalClusterList::loadFromFolder(Study& study, const AnyString& folder, A
 
                     // alias to the production cost
                     double* prodCost = cluster->productionCost;
-                    // alias to the marginal cost
-                    double marginalCost = cluster->marginalCost;
+
                     // Production cost
                     auto& modulation = cluster->modulation[thermalModulationCost];
-                    for (uint h = 0; h != cluster->modulation.height; ++h)
-                        prodCost[h] = marginalCost * modulation[h];
+                    if(cluster->costgeneration == Data::setManually)
+                    {
+                        // alias to the marginal cost
+                        double marginalCost = cluster->marginalCost;                        
+                        for (uint h = 0; h != cluster->modulation.height; ++h)
+                            prodCost[h] = marginalCost * modulation[h];
+                    }
+                    else
+                    {
+                        std::array<double, HOURS_PER_YEAR> marginalCostPerHour
+                          = cluster->marginalCostPerHourTs[0];
+                        for (uint h = 0; h != cluster->modulation.height; ++h)
+                            prodCost[h] = marginalCostPerHour[h] * modulation[h];
```

Comment:
No need for a copy here

```suggestion
                        const auto& marginalCostPerHour
                          = cluster->marginalCostPerHourTs[0];
                        for (uint h = 0; h != cluster->modulation.height; ++h)
                            prodCost[h] = marginalCostPerHour[h] * modulation[h];
```